framework/src/actions/SetupMeshCompleteAction.C

```diff
@@ -100,6 +101,9 @@ SetupMeshCompleteAction::act()
     // geometric ghosting functor and/or we have a displaced mesh
     if (_mesh->needsRemoteElemDeletion())
```

Comment:
How is this flag different from a hypothetical `distributedMesh()` flag?

Reply:
This does raise a good point ... there are two different APIs on the libMesh `MeshBase` object that tell you different things. There is `is_replicated` and  `is_serial`. The former is an indication of the class type. The latter is in indication of the current state of the class. For a `DistributedMesh`, `is_replicated` will always return `false`. However, `is_serial` will return `true` unless a user has explicitly called `set_distributed()` or has had `delete_remote_elements` called. So ... I should check the status of `is_serial` here. If `needsRemoteElemDeletion` is `true`, then I would hope that `is_serial` is also `true`. 

If `is_serial` is `true` and `needsRemoteElemDeletion` is `false` ... then if I try to `delete_remote_elements` I could possibly run across a whole slew of additional cases where previously we have not been running with a distributed mesh when we thought we were ... ok I'm going to try deleting remote elements whenever `!is_replicated() && is_serial()` and see what happens

Reply:
Sigh, this leads to 4 new failures with distributed mesh

Reply:
Would that be because we are not hitting the `is_serial()` flag? I.e. the state isn't the one we want at that point?

Reply:
No it's because there is likely more MOOSE code that doesn't actually properly support distributed mesh yet

Reply:
Cross-referencing: created #21798 based on this conversation